pkg/server/ca/manager/journal.go

```diff
@@ -25,61 +31,69 @@ const (
 	journalPEMType = "SPIRE CA JOURNAL"
 )
 
-type JournalEntries = journal.Entries
-type X509CAEntry = journal.X509CAEntry
-type JWTKeyEntry = journal.JWTKeyEntry
+type journalEntries = journal.Entries
+type x509CAEntry = journal.X509CAEntry
+type jwtKeyEntry = journal.JWTKeyEntry
+
+type journalConfig struct {
+	cat      catalog.Catalog
+	log      logrus.FieldLogger
+	filePath string
+}
 
 // Journal stores X509 CAs and JWT keys on disk as they are rotated by the
 // manager. The data format on disk is a PEM encoded protocol buffer.
 type Journal struct {
-	path string
+	config *journalConfig
 
-	mu      sync.RWMutex
-	entries *JournalEntries
+	mu                    sync.RWMutex
+	activeX509AuthorityID string
+	caJournalID           uint
+	entries               *journalEntries
 }
 
-func LoadJournal(path string) (*Journal, error) {
-	j := &Journal{
-		path:    path,
-		entries: new(JournalEntries),
-	}
-
-	pemBytes, err := os.ReadFile(path)
-	if err != nil {
-		if os.IsNotExist(err) {
+func LoadJournal(ctx context.Context, config *journalConfig) (*Journal, error) {
+	if fflag.IsSet(fflag.FlagCAJournalInDatastore) {
+		// Look for the CA journal of this server in the datastore.
+		j, err := loadJournalFromDS(ctx, config)
```

Comment:
This case should cause no problem, since any update triggers a save operation and the journal entries have the information of all the slots. The prepared CA will be stored on both the data store and disk and if the server is restarted at that time, it will load the updated journal from the data store when it starts.